### PR TITLE
feat: added dpi preview

### PR DESCRIPTION
**Pull request by jguddas**
This adds a preview with the icon pixelated to see how they look on non-high-dpi screens.

 

Example: https://github.com/jguddas/github-icon-worflow-test/pull/17#issuecomment-1536153623

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -134,6 +148,11 @@ jobs:
             ${{ steps.generate-3px-stroke-width.outputs.body }}<br/>
             </details>
             <details>
+            <summary>DPI Preview</summary>
+            ${{ steps.generate-18px-dpi-preview.outputs.body }}<br/>
+            </details>
+            <details>
```
Comment (jguddas) [inline .github/workflows/pull-request.yml:154]:
```suggestion
```

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -134,6 +148,10 @@ jobs:
             ${{ steps.generate-3px-stroke-width.outputs.body }}<br/>
             </details>
             <details>
+            <summary>DPI Preview</summary>
```
Comment (jguddas) [inline .github/workflows/pull-request.yml:151]:
```suggestion
            <summary>Min DPI Preview (18px)</summary>
```

File: .github/workflows/pull-request.yml [hunk 3]
```diff
@@ -22,6 +22,20 @@ jobs:
         uses: tj-actions/changed-files@v35
         with:
           files: icons/*.svg
+      - name: Generate 18px dpi preview
+        id: generate-18px-dpi-preview
+        run: |
+          delimiter="$(openssl rand -hex 8)"
+          echo "body<<$delimiter" >> $GITHUB_OUTPUT
+          for file in $(printf "%s\\n" icons/*.svg | shuf | head -n$(awk -F' ' '{print NF}' <<< '${{ steps.changed-files.outputs.all_changed_files }}')); do
```
Comment (jguddas) [inline .github/workflows/pull-request.yml:30]:
```suggestion
          for file in ${{ steps.changed-files.outputs.all_changed_files }}; do
```

File: .github/workflows/pull-request.yml [hunk 4]
```diff
@@ -22,6 +22,20 @@ jobs:
...
+          echo "body<<$delimiter" >> $GITHUB_OUTPUT
+          for file in ${{ steps.changed-files.outputs.all_changed_files }}; do
+            cat "$file" | # get file content
+            tr '\n' ' ' | # remove line breaks
+            sed -e 's/<svg[^>]*>/<svg>/g' | # remove attributes from svg element
+            base64 -w 0 | # encode svg
+            sed "s|.*|<img title=\"$file\" alt=\"$file\" src=\"https://lucide.dev/api/gh-icon/dpi/18/&.svg\"/> |"
+          done | tr '\n' ' ' >> $GITHUB_OUTPUT
+          echo >> $GITHUB_OUTPUT
+          echo "$delimiter" >> $GITHUB_OUTPUT
```
Comment (jguddas) [inline .github/workflows/pull-request.yml:38]:
```suggestion
      - name: Generate 18px dpi preview
        id: generate-18px-dpi-preview
        run: |
          delimiter="$(openssl rand -hex 8)"
          echo "body<<$delimiter" >> $GITHUB_OUTPUT
          for file in ${{ steps.changed-files.outputs.all_changed_files }}; do
            cat "$file" | # get file content
            tr '\n' ' ' | # remove line breaks
            sed -e 's/<svg[^>]*>/<svg>/g' | # remove attributes from svg element
            base64 -w 0 | # encode svg
            sed "s|.*|<img title=\"$file\" alt=\"$file\" src=\"https://lucide.dev/api/gh-icon/dpi/18/&.svg\"/> |"
          done | tr '\n' ' ' >> $GITHUB_OUTPUT
          echo >> $GITHUB_OUTPUT
          echo "$delimiter" >> $GITHUB_OUTPUT
      - name: Generate 24px dpi preview
        id: generate-24px-dpi-preview
        run: |
          delimiter="$(openssl rand -hex 8)"
          echo "body<<$delimiter" >> $GITHUB_OUTPUT
          for file in ${{ steps.changed-files.outputs.all_changed_files }}; do
            cat "$file" | # get file content
            tr '\n' ' ' | # remove line breaks
            sed -e 's/<svg[^>]*>/<svg>/g' | # remove attributes from svg element
            base64 -w 0 | # encode svg
            sed "s|.*|<img title=\"$file\" alt=\"$file\" src=\"https://lucide.dev/api/gh-icon/dpi/24/&.svg\"/> |"
          done | tr '\n' ' ' >> $GITHUB_OUTPUT
          echo >> $GITHUB_OUTPUT
          echo "$delimiter" >> $GITHUB_OUTPUT
```

**Review by ericfennis — COMMENTED** [review]
@jguddas Awesome work, this is really cool!

Question: What's the benefit of adding the 18px preview of the low DPI preview?

**Comment by jguddas** [comment]
> @jguddas Awesome work, this is really cool!
> 
> Question: What's the benefit of adding the 18px preview of the low DPI preview?

I think we should check that our icons work at 75% scale, using 18px high icons is super common.

File: .github/workflows/pull-request.yml [hunk 5]
```diff
@@ -134,6 +162,15 @@ jobs:
             ${{ steps.generate-3px-stroke-width.outputs.body }}<br/>
             </details>
             <details>
+            <summary>Min DPI Preview (18px)</summary>
```
Comment (jguddas) [inline .github/workflows/pull-request.yml:165]:
```suggestion
            <summary>DPI Preview (18px)</summary>
```

File: .github/workflows/pull-request.yml [hunk 6]
```diff
@@ -134,6 +162,15 @@ jobs:
             ${{ steps.generate-3px-stroke-width.outputs.body }}<br/>
             </details>
             <details>
+            <summary>DPI Preview (18px)</summary>
+            ${{ steps.generate-18px-dpi-preview.outputs.body }}<br/>
+            </details>
+            <details>
+            <details>
+            <summary>Min DPI Preview (24px)</summary>
```
Comment (jguddas) [inline .github/workflows/pull-request.yml:170]:
```suggestion
            <summary>DPI Preview (24px)</summary>
```

**Comment by ericfennis** [comment]
@jguddas Yeah but all icons are design on 24px. So 18px will probably always looks blurry. So I'm not sure if this preview is useful.

File: .github/workflows/pull-request.yml [hunk 7]
```diff
@@ -22,6 +22,34 @@ jobs:
...
+          echo "body<<$delimiter" >> $GITHUB_OUTPUT
+          for file in ${{ steps.changed-files.outputs.all_changed_files }}; do
+            cat "$file" | # get file content
+            tr '\n' ' ' | # remove line breaks
+            sed -e 's/<svg[^>]*>/<svg>/g' | # remove attributes from svg element
+            base64 -w 0 | # encode svg
+            sed "s|.*|<img title=\"$file\" alt=\"$file\" src=\"https://lucide.dev/api/gh-icon/dpi/18/&.svg\"/> |"
+          done | tr '\n' ' ' >> $GITHUB_OUTPUT
+          echo >> $GITHUB_OUTPUT
+          echo "$delimiter" >> $GITHUB_OUTPUT
```
Comment (jguddas) [inline .github/workflows/pull-request.yml:38]:
```suggestion
```

File: .github/workflows/pull-request.yml [hunk 8]
```diff
@@ -134,6 +162,15 @@ jobs:
             ${{ steps.generate-3px-stroke-width.outputs.body }}<br/>
             </details>
             <details>
+            <summary>DPI Preview (18px)</summary>
+            ${{ steps.generate-18px-dpi-preview.outputs.body }}<br/>
+            </details>
+            <details>
+            <details>
```
Comment (jguddas) [inline .github/workflows/pull-request.yml:169]:
```suggestion
```

**Review by ericfennis — APPROVED** [review]
Nice work!

**Pull request merged**